### PR TITLE
Improve the layout of the Debian packages.

### PR DESCRIPTION
**Pull request by srbaker**

### CODE REVIEW

File: packaging/installer-linux/installer-debian/src/main/resources/common/neo4j [hunk 1]
```diff
@@ -46,12 +46,12 @@ SCRIPTNAME=/etc/init.d/$NAME-service
 
 do_start()
 {
-  start-stop-daemon --chuid ${NEO_USER} --start --quiet --oknodo --pidfile $PIDFILE --exec $DAEMON -- start
+  NEO4J_BIN=/usr/bin NEO4J_CONF=${NEO4J_CONF:-} start-stop-daemon --chuid ${NEO_USER} --start --quiet --oknodo --pidfile $PIDFILE --exec $DAEMON -- start
```
Comment (srbaker) [inline packaging/installer-linux/installer-debian/src/main/resources/common/neo4j]: I don't like this. @benbc do you have suggestions for cleaning this up, and the one below it?

**Comment by srbaker** [comment]
I don't like that `neo4j-shared.sh` is in `/usr/bin` now, but I'm not sure what the better way is.

File: packaging/installer-linux/installer-debian/build.xml [hunk 2]
```diff
@@ -25,10 +25,47 @@
...
+            </copy>
+
             <replace file="${project.build.outputDirectory}/@{name}/server/conf/neo4j-wrapper.conf"
                      token="unsupported.dbms.udc.source=tarball"
                      value="unsupported.dbms.udc.source=debian"/>
 
+            <!-- Remove the files that are unnecessary, or don't make sense on Debian. -->
+            <delete file="${project.build.outputDirectory}/@{name}/server/bin/.keep"/>
+            <delete file="${project.build.outputDirectory}/@{name}/server/plugins/.keep"/>
+            <delete dir="${project.build.outputDirectory}/@{name}/server/bin/tools"/>
```
Comment (benbc) [inline packaging/installer-linux/installer-debian/build.xml]: This shouldn't be in the tarball at all. Please remove it at source rather than here.

File: packaging/installer-linux/installer-debian/build.xml [hunk 3]
```diff
@@ -25,10 +25,47 @@
...
+                    append="true">
+
+# Directory settings.
+dbms.directories.logs=/var/log/neo4j
+dbms.directories.run=/var/run/neo4j
+dbms.directories.lib=/usr/share/neo4j/lib
+dbms.directories.certificates=/var/lib/neo4j/certificates
+            </concat>
+
+            <!-- Remove the .keep directories. -->
```
Comment (benbc) [inline packaging/installer-linux/installer-debian/build.xml]: This is done elsewhere.

File: packaging/installer-linux/installer-debian/build.xml [hunk 4]
```diff
@@ -25,10 +25,47 @@
                 </fileset>
             </copy>
 
+            <copy toFile="${project.build.outputDirectory}/@{name}/server/default/neo4j">
+                <fileset dir="${project.build.sourceDirectory}/../resources/common">
+                    <include name="default-neo4j"/>
+                </fileset>
+            </copy>
```
Comment (benbc) [inline packaging/installer-linux/installer-debian/build.xml]: Can we not do the reading of /etc/default/neo4j as a separate commit? It seems like orthogonal behaviour to me and not required for the RC.

File: packaging/installer-linux/installer-debian/src/main/resources/community/debian/dirs [hunk 5]
```diff
@@ -3,7 +3,8 @@
 # (see http://people.debian.org/~seanius/policy/dbapp-policy.html/)
 var/lib/neo4j
 var/lib/neo4j/data/databases
-var/lib/neo4j/run
+var/lib/neo4j/import
+var/run/neo4j
```
Comment (benbc) [inline packaging/installer-linux/installer-debian/src/main/resources/community/debian/dirs:8]: plugins?

File: packaging/installer-linux/installer-debian/src/main/resources/community/debian/neo4j.postinst [hunk 6]
```diff
@@ -32,8 +32,8 @@ case "$1" in
         fi
       
         # directories needed for neo4j
-        chown -R $NEO_USER:adm /var/lib/neo4j /var/log/neo4j /etc/neo4j /usr/share/neo4j
-        chmod u+rwx /var/lib/neo4j /var/log/neo4j /etc/neo4j /usr/share/neo4j
+        chown -R $NEO_USER:adm /var/lib/neo4j /var/log/neo4j /etc/neo4j /usr/share/neo4j /var/run/neo4j
+        chmod u+rwx /var/lib/neo4j /var/log/neo4j /etc/neo4j /usr/share/neo4j /var/run/neo4j
```
Comment (benbc) [inline packaging/installer-linux/installer-debian/src/main/resources/community/debian/neo4j.postinst:36]: Isn't this change needed for Enterprise as well?

**Comment by benbc** [comment]
I was expecting the shell scripts to be installed to `/var/lib/neo4j/bin`, with the stuff that is invoked directly being symlinked into `/usr/bin` and then `NEO4J_BIN=/var/lib/neo4j/bin`. I think that addresses both of your comments.

**Comment by benbc** [comment]
@srbaker I think you're chmoding the new script files twice. Otherwise looks fine.
